v2.2/cluster-setup-troubleshooting.md

```diff
@@ -6,202 +6,434 @@ toc: true
+-   Unavailable Ranges: If a majority of a range's replicas are on nodes that are unavailable, then the entire range is unavailable and will be unable to process queries.
 
-### Replication error in a multi-node cluster
+**Solution:**
 
-If data is not being replicated to some nodes in the cluster, we recommend checking out the following:
+To identify under-replicated/unavailable ranges:
 
-- Ensure every node but the first was started with the `--join` flag set to the hostname and port of first node (or any other node that's successfully joined the cluster).
+1.  Access the Admin UI.
 
-    If the flag was not set correctly for a node, shut down the node and restart it with the `--join` flag set correctly. See [Stop a Node](stop-a-node.html) and [Start a Node](start-a-node.html) for more details.
+2.  On the **Cluster Overview** page, check the **Replication Status**. If the **Under-replicated ranges** or **Unavailable ranges** count is non-zero, then you have under-replicated or unavailable ranges in your cluster.
 
-- Nodes might not be able to communicate on their advertised hostnames, even though they're able to connect.
+**Add nodes to the cluster:**
 
-    You can try to resolve this by [stopping the nodes](stop-a-node.html), and then [restarting them](start-a-node.html) with the `--advertise-addr` flag set to an interface all nodes can access.
+On the Admin UI’s Cluster Overview page, check if any nodes are down. If the number of nodes down is less than (n-1)/2, then that is most probably the cause of the under-replicated/unavailable ranges. Add nodes to the cluster such that the cluster has the required number of nodes to replicate ranges properly.
 
-- Check the [logs](debug-and-error-logs.html) for each node for further detail, as well as these common errors:
+If you still under-replicated/unavailable ranges on the Cluster Overview page, investigate further:
 
-  - `connection refused`: [Troubleshoot your network](#networking-troubleshooting).
-  - `not connected to cluster` or `node <id> belongs to cluster...`: See [Node Won't Join Cluster](#node-wont-join-cluster) on this page.
+1.  On the Admin UI, click the gear icon on the left-hand navigation bar to access the **Advanced Debugging** page.
+2.  On the Advanced Debugging page, click **Problem Ranges**.
+3.  In the **Connections** table, identify the node with the under-replicated/unavailable ranges and click the node ID in the Node column.
+4.  To view the **Range Report** for a range, click on the range number in the **Under-replicated (or slow)** table or **Unavailable** table.
+5. On the Range Report page, scroll down to the **Simulated Allocator Output** section. The table contains an error message which explains the reason for the under-replicated range. Follow the guidance in the message to resolve the issue. If you need help understanding the error or the guidance, file an issue.
```

Comment:
"...file an issue. Please be sure to include the full range report and error message when you submit it."

Reply:
Done.